File: DatosCompuesto.java
//Arreglo de una dimension
import java.util.Scanner;

public class DatosCompuesto {
    public static void main(String[] args) {
        final int NUM = 8;
        int[] nums = new int[NUM];
        int total = 0;

        Scanner scanner = new Scanner(System.in);
        for (int i = 0; i < NUM; i++) {
            System.out.println("Por favor, introduzca el número: ");
            nums[i] = scanner.nextInt();
            total += nums[i];
        }
        System.out.println("El total de números es: " + total);
        scanner.close();
    }
}
 
//Arreglo de dos dimensiones

public class DatosCompuesto {
    public static void main(String[] args) {
        // Arreglo de una dimension
        final int NUM = 8;
        int[] nums = new int[NUM];
        int total = 0;

        Scanner scanner = new Scanner(System.in);
        for (int i = 0; i < NUM; i++) {
            System.out.println("Por favor, introduzca el número: ");
            nums[i] = scanner.nextInt();
            total += nums[i];
        }
        System.out.println("El total de números es: " + total);
        scanner.close();

        // Arreglo de dos dimensiones
        final int N = 2;
        final int M = 4;
        float[][] discos = new float[N][M];
        scanner = new Scanner(System.in);
        for (int fila = 0; fila < N; fila++) {
            for (int col = 0; col < M; col++) {
                discos[fila][col] = scanner.nextFloat();
            }
        }

        for (int fila = 0; fila < N; fila++) {
            System.out.print("Precio fila " + fila + "\t");
            for (int col = 0; col < M; col++) {
                System.out.print(discos[fila][col] + "\t");
            }
            System.out.println();
        }
    }
}
//Arrays como parametros en funciones
public class ArrayParams {
    public static final int LONG = 100;

    public static void leerArray(double[] a, int[] n) {
        n[0] = 0;
        System.out.println("Introduzca datos. Para terminar pulsar 0:");
        for (int i = 0; i < LONG; i++) {
            System.out.print(i + " : ");
        }
    }

    public static void imprimirArray(double[] a, int n) {
        System.out.println("El array tiene " + n + " elementos son:");
        for (int i = 0; i < n; i++) {
            System.out.println("\t" + i + " : " + a[i]);
        }
    }

    public static void main(String[] args) {
        double[] a = new double[LONG];
        int[] n = new int[1];
        leerArray(a, n);
        System.out.print("Ingrese un número (0 para terminar): ");
        a[n[0]] = Double.parseDouble(System.console().readLine());
        if (a[n[0]] == 0) {
            System.out.println("Fin del programa");
            return;
        }
        System.out.println("El array tiene " + n[0] + " elementos son:");
        imprimirArray(a, n[0]);
    }
}
//Estructura de Objetos
class Empleado {
    // Atributos de la clase Empleado
    int id;
    String nombre;
    String apellido;
    float salario;
    int edad;

    // Constructor para inicializar los atributos de la clase
    public Empleado(int id, String nombre, String apellido, float salario, int edad) {
        this.id = id;
        this.nombre = nombre;
        this.apellido = apellido;
        this.salario = salario;
        this.edad = edad;
    }

    // Método para mostrar información del empleado
    public void mostrarInformacion() {
        System.out.println("ID: " + id);
        System.out.println("Nombre: " + nombre + " " + apellido);
        System.out.println("Salario: " + salario);
        System.out.println("Edad: " + edad);
    }
}

public class Main {
    public static void main(String[] args) {
        // Crear un objeto Empleado
        Empleado empleado = new Empleado(1, "Juan", "Pérez", 2500.50f, 30);

        // Mostrar la información del empleado
        empleado.mostrarInformacion();
    }
}


//Enumeracion

public class EnumDemo {
    enum Color {
        BLANCO, AZUL, VERDE, ROJO
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.println("Colores disponibles: " + Color.values().length);
        for (Color c : Color.values()) {
            System.out.println(c); // Muestra los valores del enum
        }

        System.out.println("Seleccione un color: ");
        String colorInput = scanner.next().toUpperCase(); 

        try {
            Color rotulador = Color.valueOf(colorInput);
            System.out.println("El color seleccionado es: " + rotulador);

            switch (rotulador) {
                case ROJO:
                    System.out.println("El rotulador es rojo");
                    break;
                case VERDE:
                    System.out.println("El rotulador es verde");
                    break;
                case AZUL:
                    System.out.println("El rotulador es azul");
                    break;
                case BLANCO:
                    System.out.println("El rotulador es blanco");
                    break;
            }
        } catch (IllegalArgumentException e) {
            System.out.println("Error: Color no válido. Asegúrese de usar uno de los valores disponibles.");
        } finally {
            scanner.close();
        }
    }
}

File: Datoscompuesto.js
//Arreglo de una dimensión 
function datosCompuesto() {
    const NUM = 8; 
    const nums = new Array(NUM).fill(0);
    let total = 0; 

    for (let i = 0; i < NUM; i++) {
        nums[i] = parseInt(prompt("Por favor, introduzca el número:"), 10); 
        total += nums[i]; 
    }

    console.log("El total de números es:", total);
}
datosCompuesto();

//Arreglo de dos dimensiones 
function arregloDosDimensiones() {
    const N = 2; 
    const M = 4; 
    const discos = Array.from({ length: N }, () => new Array(M).fill(0)); 

    for (let fila = 0; fila < N; fila++) {
        for (let col = 0; col < M; col++) {
            discos[fila][col] = parseFloat(prompt(`Introduzca el valor para discos[${fila}][${col}]:`)); 
        }
    }

    for (let fila = 0; fila < N; fila++) {
        let filaString = `Precio fila ${fila}\t`; 
        for (let col = 0; col < M; col++) {
            filaString += `${discos[fila][col]}\t`; 
        }
        console.log(filaString); 
    }
}
arregloDosDimensiones();

//Arreglo como parámetro
function leerArray(a, n) {
    console.log("Introduzca datos. Para terminar, pulsar 0:");
    n[0] = 0; 
    while (true) {
        let valor = parseFloat(prompt(`Dato ${n[0]}: `)); 
        if (valor === 0) break; 
        a[n[0]] = valor; 
        n[0] += 1; 
    }
}

function imprimirArray(a, n) {
    console.log("El arreglo tiene", n, "elementos:");
    for (let i = 0; i < n; i++) {
        console.log(`\t${i} : ${a[i]}`); 
    }
}

function main() {
    const LONG = 100; 
    const a = new Array(LONG).fill(0); 
    const n = [0]; 

    leerArray(a, n); 
    imprimirArray(a, n[0]); 
}
main();

//Estructura de Objetos

class Empleado {
    constructor(id, nombre, apellido, salario, edad) {
        this.id = id;
        this.nombre = nombre;
        this.apellido = apellido;
        this.salario = salario;
        this.edad = edad;
    }

    mostrarInformacion() {
        console.log("ID:", this.id);
        console.log("Nombre:", this.nombre, this.apellido);
        console.log("Salario:", this.salario);
        console.log("Edad:", this.edad);
    }
}


function main() {
  
    const empleado = new Empleado(1, "Juan", "Pérez", 2500.50, 30);
    
    empleado.mostrarInformacion();
}

main();

//Enumeraciones

const Color = {
    BLANCO: 'BLANCO',
    AZUL: 'AZUL',
    VERDE: 'VERDE',
    ROJO: 'ROJO'
};

function main() {
    console.log("Colores disponibles:", Object.values(Color)); 

    const colorInput = prompt("Seleccione un color (BLANCO, AZUL, VERDE, ROJO): ").toUpperCase(); 
    
    try {
        const rotulador = Color[colorInput]; 
        console.log("El color seleccionado es:", rotulador);

        
        if (rotulador === Color.ROJO) {
            console.log("El rotulador es rojo");
        } else if (rotulador === Color.VERDE) {
            console.log("El rotulador es verde");
        } else if (rotulador === Color.AZUL) {
            console.log("El rotulador es azul");
        } else if (rotulador === Color.BLANCO) {
            console.log("El rotulador es blanco");
        }
    } catch (error) {
        console.log("Error: Color no válido. Asegúrese de usar uno de los valores disponibles.");
    }
}


main();
